core/pkg/artifacts/linker.go

```diff
@@ -60,21 +59,20 @@ func (al *ArtifactLinker) Link() error {
 			nil,
 		)
 	default:
-		return fmt.Errorf(
+		err = fmt.Errorf(
 			"LinkArtifact: %s,"+
 				" error: artifact must have either server id or client id",
 			portfolioName,
 		)
 	}
 
 	if err != nil {
-		return fmt.Errorf(
-			"LinkArtifact: %s, error: %v, response: %v",
+		err = fmt.Errorf(
```

Comment:
If you hit the `default` case above, the error will be

"LinkArtifact: portfolioName, error: LinkArtifact: portfolioName, error: artifact must ..."